tests/snapshots/tests.integration.test_propagation.test_single_span_sampling_tags_are_removed_when_entire_trace_is_sampled.json

```diff
@@ -0,0 +1,23 @@
+[[
```

Comment:
Wouldn't this span be dropped by the Datadog agent, when it should be kept, since the single span sampling metrics are removed?

Reply:
Oh, I didn't know that the agent cared about `_dd.p.dm`. Did we confirm that?

Reply:
I haven't - I took it as a given based on some explanation I got from Brett. I'll see if I can figure out how to add that piece of knowledge to a test.

Reply:
@ZStriker19 Your comment here prompted me to take another look at the various interactions of types of sampling, and I fixed the tests in this pull request to be much clearer about the expected behavior.